geode-core/src/main/java/org/apache/geode/security/PostProcessor.java

```diff
@@ -44,6 +47,9 @@ default void init(Properties securityProps) {}
    */
   Object processRegionValue(Object principal, String regionName, Object key, Object value);
 
+  Collection<Object> processQueryResult(Object principal, Query query, Collection<String> regions,
```

Comment:
I don't think we want to add this method in the interface. For query,  essentially it's still getting value from region. Customers should not be burdened with another method to implement if they just want to manipulate the data. See how query result get post processed in gfsh queries.